source/Octopus.Client/LocationChecker.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Win32;
+using Octopus.Client.Logging;
+
+namespace Octopus.Client
+{
+#if FULL_FRAMEWORK
```

Comment:
if you move this down inside the `CheckAssemblyLocation` method, you can ditch the `#if`s in the other locations

Reply:
I decided to go this way so the whole thing just doesn't exist in the NETCore world

Reply:
@slewis74 you could move the conditional compile directive to the top of the file